apps/dapp/src/app/api/storage/route.ts

```diff
@@ -0,0 +1,38 @@
+import { Readable } from 'stream';
+import pinningService from '@/storage';
+import { FileError } from '@/types';
+
+const { uploadFile, uploadJSON } = pinningService();
+
+export async function POST(request: Request): Promise<Response> {
+  let result;
+
+  const contentType = request.headers.get('content-type');
+
+  if (contentType?.includes('multipart/form-data')) {
+    const formData = await request.formData();
+    const file = formData.get('file') as Blob;
+    const name = formData.get('name') as string;
+
+    if (file instanceof Blob && typeof name === 'string') {
+      const buffer = Buffer.from(await file.arrayBuffer());
+
+      const readableStream = new Readable({
+        read() {
+          this.push(buffer);
+          this.push(null);
```

Comment:
Why do you push null?

Reply:
Pushing null signals the end of the stream